### PR TITLE
chore(libbpf): install libbpf for development from source

### PR DESCRIPTION
**Pull request by vimalk78**
This PR installs libbpf (`v1.2.0`) from source. 

Installing from source shields us from differences in the way libbpf is provided in various distributions. e.g. Ubuntu 22 LTS the package name to be installed is different from fedora/ubi/rhel. Also the provided version is also not consistent across distributions.

### CODE REVIEW

**Comment by rootfs** [comment]
please run `npm run prepare` and also update verify.sh to verify libbpf is installed.

**Comment by vimalk78** [comment]
Cc: @rootfs @SamYuan1990

File: index.js [hunk 1]
```diff
@@ -38,8 +38,17 @@ function installXgboost(artifacts_version, xgboost_version) {
 }
 
 function installLibbpf() {
-  core.info(`libbpf`);
-  executeCommand("sudo apt-get install -y libbpf-dev", "fail to install libbpf related package");
+  core.info(`installing libbpf v1.2.0 from source`);
```
Comment (rootfs) [inline index.js:41]: get the libbpf version from env (as other pkgs)

Comment (vimalk78) [reply index.js]: updated

Comment (vimalk78) [reply index.js]: can you pls run the tests?

File: index.js [hunk 2]
```diff
@@ -37,9 +37,18 @@ function installXgboost(artifacts_version, xgboost_version) {
...
-function installLibbpf() {
-  core.info(`libbpf`);
-  executeCommand("sudo apt-get install -y libbpf-dev", "fail to install libbpf related package");
+function installLibbpf(libbpf_version) {
+  core.info(`installing libbpf version ` + libbpf_version + ` from source`);
+  executeCommand("mkdir -p temp-libbpf");
+  executeCommand("cd temp-libbpf");
+  executeCommand("git clone https://github.com/libbpf/libbpf");
+  executeCommand("git checkout "+ libbpf_version);
+  executeCommand("cd libbpf/src");
```
Comment (sthaha) [inline index.js:46]: would this actually switch directory? wouldn't the directory be back to current once the command finishes executing? IOW, doesn't this just open a shell, cd and then close the shell?

File: index.js [hunk 3]
```diff
@@ -37,9 +37,18 @@ function installXgboost(artifacts_version, xgboost_version) {
...
 
-function installLibbpf() {
-  core.info(`libbpf`);
-  executeCommand("sudo apt-get install -y libbpf-dev", "fail to install libbpf related package");
+function installLibbpf(libbpf_version) {
+  core.info(`installing libbpf version ` + libbpf_version + ` from source`);
+  executeCommand("mkdir -p temp-libbpf");
+  executeCommand("cd temp-libbpf");
+  executeCommand("git clone https://github.com/libbpf/libbpf");
+  executeCommand("git checkout "+ libbpf_version);
```
Comment (sthaha) [inline index.js:45]:
```suggestion
```

File: index.js [hunk 4]
```diff
@@ -37,9 +37,18 @@ function installXgboost(artifacts_version, xgboost_version) {
...
+  executeCommand("cd libbpf/src");
+  executeCommand("sudo make install_headers");
+  executeCommand("sudo make install_uapi_headers");
+  executeCommand("sudo prefix=/usr BUILD_STATIC_ONLY=y make install");
```
Comment (sthaha) [inline index.js:49]: `PREFIX` 
 
https://github.com/libbpf/libbpf/blob/20c0a9e3d7e7d4aeb283eae982543c9cacc29477/src/Makefile#L97

but `PREFIX` is already set to `/usr`

**Comment by vimalk78** [comment]
@rootfs can i pls get rights to run tests and request for reviews?

**Review by vprashar2929 — APPROVED** [review]
LGTM

**Comment by vimalk78** [comment]
@rootfs all tests have passed. if no more comments can we merge it? i want to update kepler unit test to use this.

File: index.js [hunk 5]
```diff
@@ -37,9 +37,15 @@ function installXgboost(artifacts_version, xgboost_version) {
...
-  core.info(`libbpf`);
-  executeCommand("sudo apt-get install -y libbpf-dev", "fail to install libbpf related package");
+function installLibbpf(libbpf_version) {
+  core.info(`installing libbpf version ` + libbpf_version + ` from source`);
+  executeCommand("sudo apt install libelf-dev", "failed to install libelf-dev");
+  executeCommand("mkdir -p temp-libbpf");
+  executeCommand("cd temp-libbpf && git clone -b " + libbpf_version + " https://github.com/libbpf/libbpf");
+  executeCommand("cd temp-libbpf/libbpf/src && sudo make install_headers", "failed to install libbpf headers");
+  executeCommand("cd temp-libbpf/libbpf/src && sudo make install_uapi_headers", "failed to install libbpf headers");
+  executeCommand("cd temp-libbpf/libbpf/src && sudo prefix=/usr BUILD_STATIC_ONLY=y make install", "failed to install libbpf archive library");
```
Comment (sthaha) [inline index.js:47]: PREFIX instead of `prefix` ?

**Review by sthaha — CHANGES_REQUESTED** [review]
requesting change to either remove use of `prefix` or use `PREFIX` 
https://github.com/libbpf/libbpf/blob/20c0a9e3d7e7d4aeb283eae982543c9cacc29477/src/Makefile#L97

File: package-lock.json
Comment (sthaha) [inline package-lock.json:1]: @vimalk78 this just caught my eye, why are we deleting the lock file?

**Review by sthaha — CHANGES_REQUESTED** [review]
Requesting changes to update the lockfile instead of removing it.

File: package-lock.json
Comment (vimalk78) [reply package-lock.json:1]: no, i updated the `@actions/core` from `1.10.0` to `1.10.1`. executed `npm` and .lock file was updated. It is not deleted.

**Comment by vimalk78** [comment]
@rootfs can we make a new release of `kepler-action` for this PR? it is needed to be used kepler unit test https://github.com/sustainable-computing-io/kepler/pull/1097

**Pull request merged**